tf/tfprocess.py

```diff
@@ -523,134 +547,113 @@ def calculate_test_summaries(self, test_batches, steps):
+            work_weights = None
+            if weights.shape.ndims == 4:
+                # Convolution weights need a transpose
+                #
+                # TF (kYXInputOutput)
+                # [filter_height, filter_width, in_channels, out_channels]
+                #
+                # Leela/cuDNN/Caffe (kOutputInputYX)
+                # [output, input, filter_size, filter_size]
+                work_weights = tf.transpose(a=weights, perm=[3, 2, 0, 1])
+            elif weights.shape.ndims == 2:
+                # Fully connected layers are [in, out] in TF
+                #
+                # [out, in] in Leela
+                #
+                work_weights = tf.transpose(a=weights, perm=[1, 0])
+            else:
+                # batch renorm has extra weights, but we don't know what to do with them.
+                if 'renorm' in weights.name:
+                    continue
+                # renorm has variance, but it is not the primary source of truth
+                if 'variance:' in weights.name and self.renorm_enabled:
+                    continue
+                # Renorm has moving stddev not variance, undo the transform to make it compatible.
+                if 'stddev:' in weights.name:
+                    all_tensors.append(tf.math.square(weights) - 1e-5)
+                    continue
+                # Biases, batchnorm etc
+                # pb expects every batch norm to have gammas, but not all of our
+                # batch norms have gammas, so manually add pretend gammas.
+                if 'beta:' in weights.name and not last_was_gamma:
+                    all_tensors.append(tf.ones_like(weights))
+                work_weights = weights.read_value()
+            all_tensors.append(work_weights)
+            last_was_gamma = 'gamma:' in weights.name
+
+        # HACK: model weights ordering is some kind of breadth first traversal,
+        # but pb expects a specific ordering which BFT is not a match for once
+        # we get to the heads. Apply manual permutation.
+        # This is fragile and at minimum should have some checks to ensure it isn't breaking things.
+        #TODO: also support classic policy head as it has a different set of layers and hence changes the permutation.
+        permuted_tensors = [w for w in all_tensors]
+        permuted_tensors[-5] = all_tensors[-11]
+        permuted_tensors[-6] = all_tensors[-12]
+        permuted_tensors[-7] = all_tensors[-13]
+        permuted_tensors[-8] = all_tensors[-14]
+        permuted_tensors[-9] = all_tensors[-16]
+        permuted_tensors[-10] = all_tensors[-5]
+        permuted_tensors[-11] = all_tensors[-6]        
+        permuted_tensors[-12] = all_tensors[-7]        
+        permuted_tensors[-13] = all_tensors[-8]        
+        permuted_tensors[-14] = all_tensors[-9]        
+        permuted_tensors[-15] = all_tensors[-10]        
+        permuted_tensors[-16] = all_tensors[-15]        
```

Comment:
There is lot of trailing whitespace here.

Reply:
I think I fixed it...